### PR TITLE
chore: compatablility with KILT parachain

### PR DESCRIPTION
**Pull request by weichweich**
## fixes KILTProtocol/ticket#886

Companion PR to https://github.com/KILTprotocol/kilt-parachain/pull/5

 This depends on @rflechtner changes in #322. 

## Todo:

- [ ] integration tests for a parachain?

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by weichweich** [comment]
Contains changes from https://github.com/KILTprotocol/sdk-js/pull/322

File: packages/core/src/blockchainApiConnection/BlockchainApiConnection.ts [hunk 1]
```diff
@@ -9,43 +9,38 @@
...
 
 import { ApiPromise, WsProvider } from '@polkadot/api'
-import { RegistryTypes } from '@polkadot/types/types'
 import Blockchain from '../blockchain/Blockchain'
+import MASHNET_TYPES from './types/mashnet'
+import PARACHAIN_TYPES from './types/parachain'
 
 export const DEFAULT_WS_ADDRESS =
-  process.env.DEFAULT_WS_ADDRESS || 'ws://127.0.0.1:9944'
+  process.env.DEFAULT_WS_ADDRESS || 'ws://127.0.0.1:9977'
```
Comment (tjwelde) [inline packages/core/src/blockchainApiConnection/BlockchainApiConnection.ts:17]: why this change?

File: packages/core/src/identity/Identity.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ export default class Identity {
    * ```
    */
   public static buildFromSeed(seed: Uint8Array): Identity {
-    const keyring = new Keyring({ type: 'ed25519' })
+    const keyring = new Keyring({ type: 'sr25519' })
```
Comment (tjwelde) [inline packages/core/src/identity/Identity.ts:126]: That is a bigger change, we should maybe do separately, or is this needed for parachains?

File: packages/core/src/__integrationtests__/Balance.spec.ts [hunk 3]
```diff
@@ -80,7 +80,7 @@ describe('when there is a dev chain with a faucet', () => {
     expect(balanceBefore.sub(balanceAfter).gt(MIN_TRANSACTION)).toBeTruthy()
     expect(balanceIdent.toNumber()).toBe(MIN_TRANSACTION.toNumber())
     expect(funny).toBeCalled()
-  }, 30_000)
+  }, 60_000)
```
Comment (tjwelde) [inline packages/core/src/__integrationtests__/Balance.spec.ts:83]: Why more?

File: packages/core/src/blockchainApiConnection/BlockchainApiConnection.ts (hunk 1 above)
Comment (weichweich) [reply packages/core/src/blockchainApiConnection/BlockchainApiConnection.ts:17]: Parachains have two ws ports. 9944 is for the relay chain, 9977 is for the parachain.

We connect to the parachain...

File: packages/core/src/identity/Identity.ts (hunk 2 above)
Comment (weichweich) [reply packages/core/src/identity/Identity.ts:126]: This is only needed to have balances on the parachain since the endowed accounts are sr25519 accounts. Would do that in this PR either, but this was needed to test with the parachain.

File: packages/core/src/__integrationtests__/Balance.spec.ts (hunk 3 above)
Comment (weichweich) [reply packages/core/src/__integrationtests__/Balance.spec.ts:83]: Tests will take more time with the parachain. Blocktime is 12s not 6s. Also the blocktime is not constant. It takes longer sometimes.

**Comment by weichweich** [comment]
Parachain types might be included in the latest polkadot-js. The separate types might not be needed anymore.

**Comment by tjwelde** [comment]
#322 is now merged, so this PR can just concentrate on the compatibility to the kilt parachain

**Comment by weichweich** [comment]
this can be closed. The most important change was the update the rest is out dated and needs to be checked again...
